lasagne/tests/layers/test_noise.py

```diff
@@ -60,6 +65,11 @@ def test_get_output_for_p_02(self, layer_p_02):
         assert 0.9 < result_eval.mean() < 1.1
         assert (numpy.round(numpy.unique(result_eval), 2) == [0., 1.25]).all()
 
+    def test_get_output_for_p_float32_dtype(self, layer_p_float32):
+        input = theano.shared(numpy.ones((100, 100), dtype=numpy.float32))
+        result = layer_p_float32.get_output_for(input)
+        assert result.dtype == input.dtype
```

Comment:
As `layer_p_float32` is only used once, I'd prefer not to define it and write the test directly as:
```python
def test_get_output_for_float32_p(self):
    floatX_ = theano.config.floatX
    theano.config.floatX = 'float32'
    from lasagne.layers.noise import DropoutLayer
    layer = DropoutLayer((None, 100), p=numpy.float32(0.5))
    input = T.matrix()
    assert layer.get_ouput_for(input).dtype == input.dtype
    theano.config.floatX = floatX_
```